riscv/zicfiss.h

```diff
@@ -0,0 +1,29 @@
+// See LICENSE for license details.
+
+#ifndef _RISCV_ZICFISS_H
+#define _RISCV_ZICFISS_H
+
+#define xSSE() \
+  ((STATE.prv != PRV_M) && get_field(STATE.menvcfg->read(), MENVCFG_SSE) && \
+  p->extension_enabled('S') && \
+  ((STATE.v && get_field(STATE.henvcfg->read(), HENVCFG_SSE)) || !STATE.v) && \
+  (((STATE.prv == PRV_U) && get_field(STATE.senvcfg->read(), SENVCFG_SSE)) || (STATE.prv != PRV_U)))
+
+#define PUSH_VALUE_TO_SS(value) \
+  reg_t push_ssp_addr = STATE.ssp->read() - xlen / 8; \
+  if (xlen == 32) \
+    MMU.ss_store<uint32_t>(push_ssp_addr, value); \
```

Comment:
Best practice in macros is to always surround params with parens: `(value)`

Also since you're using `value` twice, and it might possibly have side effects, it would be best to assign it to a local variable first. See `#define WRITE_REG` in `decode_macros.h` for example.

Reply:
Thank you for your reminder.